### PR TITLE
feat(ecr): create first version of registry, docker image and ecs

### PR DESCRIPTION
**Pull request by Lmam22**
Create a first version of registry, docker image and ecs

### CODE REVIEW

File: locals.tf [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# locals {
```
Comment (jose-sal-y-rosas) [inline locals.tf:1]: Please remove this file if all content is commented

File: main.tf [hunk 2]
```diff
@@ -6,3 +6,11 @@ module "vpc_aws" {
   database_subnet_cidrs = var.database_subnet_cidrs
   azs                   = var.azs
 }
+
+module "ecr-repo" {
```
Comment (jose-sal-y-rosas) [inline main.tf:10]: Please use a generic name such as 'container_registry'

File: providers.tf [hunk 3]
```diff
@@ -10,3 +10,11 @@ terraform {
 provider "aws" {
   region = var.aws_region
 }
+
+# provider "docker" {
```
Comment (jose-sal-y-rosas) [inline providers.tf:14]: Please remove this section if it not used

**Review by jose-sal-y-rosas — CHANGES_REQUESTED** [review]
Please review feedback

**Review by pramirezma — COMMENTED** [review]
At the end, you are using Terraform Cloud for the variables?

**Comment by Lmam22** [comment]
> At the end, you are using Terraform Cloud for the variables?

Just for AWS_ACCESS_KEY_ID, AWS_SECRET_KEY_ID and SESSION_TOKEN

File: locals.tf (hunk 1 above)
Comment (Lmam22) [reply locals.tf:1]: Ok

**Review by Lmam22 — COMMENTED** [review]
done

**Comment by Lmam22** [comment]
Helllo World with ECS and Fargate: http://example-lb-110786881.us-east-1.elb.amazonaws.com/
image

**Comment by pramirezma** [comment]
Is not a bad idea just do it in only one part, just to not "overcomplicate" the setup

**Review by pramirezma — APPROVED** [review]

**Comment by Lmam22** [comment]
http://example-lb-208770439.us-east-1.elb.amazonaws.com/

image

**Comment by pramirezma** [comment]
Remove the comments if there is no need to have them

File: modules/ecr/variables.tf [hunk 4]
```diff
@@ -0,0 +1,4 @@
+variable "ecr_name" {}
+variable "image_mutability" {}
+variable "encrypt_type" {}
+variable "aws_region" {}
```
Comment (jose-sal-y-rosas) [inline modules/ecr/variables.tf:4]: Are these variables don't required a description?

File: modules/ecs/alb.tf [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+################CREATE APPLICATION LOAD BALANCER
+resource "aws_lb" "default" {
+  name = "example-lb"
+  subnets         = [var.public_subnets_ids[0], var.public_subnets_ids[1]] #public subnets
+  security_groups = [aws_security_group.lb.id]
+
+
+  # count                   = length(var.public_subnets_ids) #2
```
Comment (jose-sal-y-rosas) [inline modules/ecs/alb.tf:32]: Please remove commented lines

File: modules/ecs/cluster.tf [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+  network_configuration {
+    security_groups = [aws_security_group.hello_world_task.id]
+    subnets         = [var.private_subnets_ids[0], var.private_subnets_ids[1]] #private subnets
+  }
+
+  load_balancer {
+    target_group_arn = aws_lb_target_group.hello_world.id
+    container_name   = "chupito-front"
+    #container_port   = 3000
```
Comment (jose-sal-y-rosas) [inline modules/ecs/cluster.tf:21]: Please remove commented lines

File: modules/ecs/ecs_task.tf [hunk 7]
```diff
@@ -0,0 +1,52 @@
+resource "aws_ecs_task_definition" "hello_world" {
+  family                   = "chupito-front"
+  network_mode             = "awsvpc"
+  requires_compatibilities = ["FARGATE"]
+  cpu                      = 1024
+  memory                   = 2048
+
+
+  #"image": "${aws_ecr_repository.ecr.repository_url}",
```
Comment (jose-sal-y-rosas) [inline modules/ecs/ecs_task.tf:9]: Please remove commented line

File: modules/ecs/ecs_task.tf [hunk 8]
```diff
@@ -0,0 +1,52 @@
+resource "aws_ecs_task_definition" "hello_world" {
```
Comment (jose-sal-y-rosas) [inline modules/ecs/ecs_task.tf:1]: hello world as name of the resource?

File: modules/ecs/ecs_task.tf [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+  execution_role_arn = aws_iam_role.ecsTaskExecutionRole.arn
+}
+
+resource "aws_security_group" "hello_world_task" {
+  name   = "task-security-group"
+  vpc_id = var.vpc_id
+
+  ingress {
+    protocol        = "tcp"
+    #from_port       = 3000
```
Comment (jose-sal-y-rosas) [inline modules/ecs/ecs_task.tf:39]: Please remove all commented lines

**Review by jose-sal-y-rosas — CHANGES_REQUESTED** [review]
Please review feedback

**Comment by Lmam22** [comment]
http://alb-chupito-1636170153.us-east-1.elb.amazonaws.com/

image

**Review by pramirezma — APPROVED** [review]

**Review by pamelaluquemck — APPROVED** [review]

File: modules/ecr/variables.tf (hunk 4 above)
Comment (jose-sal-y-rosas) [reply modules/ecr/variables.tf:4]: Please add the description for these variables.

**Review by jose-sal-y-rosas — CHANGES_REQUESTED** [review]
Please review feedback

**Review by jose-sal-y-rosas — APPROVED** [review]
👍🏼

**Pull request merged**